### PR TITLE
Issue #9895: Expose getExperimentBranches in NimbusClientInterface

### LINKED ISSUE #9895: Expose getExperimentBranches in NimbusClientInterface

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #9895 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt
Comment (jhugman) [inline components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:269]: This needs a `withCatchAll` wrapping.

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt
Comment (jhugman) [inline components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:76]: Nice!

**Review by jhugman — APPROVED** [review]
This requires a catchall wrapping, but we shouldn't hold up on another review cycle.

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt
Comment (travis79) [inline components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:268]: @jhugman Does this need a `getExperimentBranchesOnThisThread` and similar to the other API functions and get called on the dbScope since it is calling `get_all_experiments` and thus performing a `db.read()`?

**Review by travis79 — CHANGES_REQUESTED** [review]

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt
Comment (gabrielluong) [reply components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt]: Waiting on a response to this before landing.

Comment (jhugman) [reply components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt]: This is protected by an `@WorkerThread`; if we want to be able to call it from any/main thread, I would suggest we'd add it to the `DatabaseCache` in the Nimbus-SDK.

**Review by travis79 — APPROVED** [review]

**Pull request merged**
